### PR TITLE
Rovpp input

### PR DESCRIPTION
**Pull request by Iuconnu**
This compiles but there are some polymorphism issues that need to be sorted out. Specifically the ases map in ASGraph is <uin32_t, AS*> and I'm uncertain how to correctly reassign the type to <uint32_t, ROVppAS*>.

### CODE REVIEW

File: Extrapolator.cpp [hunk 1]
```diff
@@ -33,6 +33,18 @@ int g_loop = 0;
 int g_ts_tb = 0;
 int g_broken_path = 0;
 
+// ROV++ constructor
+Extrapolator::Extrapolator(ASGraph *g,
```
Comment (c-morris) [inline Extrapolator.cpp]: This is much cleaner than the approach I was taking. :+1:

File: ROVppExtrapolator.cpp [hunk 2]
```diff
@@ -24,21 +24,15 @@
...
-ROVppExtrapolator::ROVppExtrapolator(bool invert_results,
-                           bool store_depref,
-                           std::string a,
-                           std::string r,
-                           std::string i,
-                           std::string d,
+ROVppExtrapolator::ROVppExtrapolator(std::string r,
+                           std::string e,
+                           std::string f,
+                           std::string g,
```
Comment (c-morris) [inline ROVppExtrapolator.cpp]: This makes sense, I like removing the parameters that only have one value that works.

File: ROVppAS.h [hunk 3]
```diff
@@ -25,15 +25,19 @@
 #define ROVPPAS_H
 
 #include "AS.h"
+#include "ROVppAnnouncement.h"
 
 struct ROVppAS : public AS {
+    std::vector<uint32_t> policy_array;
```
Comment (c-morris) [inline ROVppAS.h]: Calling this variable `policy_array` when its type is a vector might become confusing later on. Something like `policies` or `policy_vector` would be better.

File: ROVppASGraph.cpp [hunk 4]
```diff
@@ -23,7 +23,48 @@
...
+    // TODO should take a parameter table name
+    // TODO currently takes only a single policy
+    R = querier->select_AS_flags();
+    for (pqxx::result::const_iterator c = R.begin(); c != R.end(); ++c) {
+        auto search = ases->find(c["asn"].as<uint32_t>());
+        if (search != ases->end()) {
+            // TODO Fix polymorphism here
+            //search->second->policy_array.insert(c["as_type"].as<uint32_t>());
+            //Polymorphic function?
+            search->second->add_policy(c["as_type"].as<uint32_t>());
```
Comment (c-morris) [inline ROVppASGraph.cpp]: Is the TODO Fix polymorphism here done? It looks like you have it done here, so if it's done let's remove the comment.

Comment (Iuconnu) [reply ROVppASGraph.cpp]: Yeah, it should be fixed now.

**Pull request merged**